### PR TITLE
Azure PowerShell  - Clear context at process level

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzurePowerShellV2/task.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "version": {
         "Major": 2,
         "Minor": 1,
-        "Patch": 4
+        "Patch": 5
```
Comment (kmkumaran) [inline Tasks/AzurePowerShellV2/task.json]: Can you check if these versions are ok to be published and consumed for 143 branch?

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 [hunk 2]
```diff
@@ -63,8 +63,12 @@ function Initialize-AzureSubscription {
     if ($Endpoint.Auth.Scheme -eq 'ServicePrincipal' -and !$script:azureModule -and (Get-Command -Name "Clear-AzureRmContext" -ErrorAction "SilentlyContinue")) {
         Write-Host "##[command]Clear-AzureRmContext -Scope Process"
         $null = Clear-AzureRmContext -Scope Process
-        Write-Host "##[command]Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue"
-        $null = Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
+    }
```
Comment (kmkumaran) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: Area you planning to port #9327 separately? Can we get all required changes in single Ask mode?

Comment (vincentdass) [reply Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: this file is not present in M143.

File: Tasks/AzurePowerShellV2/task.json (hunk 1 above)
Comment (vincentdass) [reply Tasks/AzurePowerShellV2/task.json]: M143 version `2.1.4`
M144 version `2.1.6`
We are good to add this version.

File: Tasks/AzurePowerShellV3/task.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "version": {
         "Major": 3,
         "Minor": 1,
-        "Patch": 11
+        "Patch": 12
```
Comment (vincentdass) [inline Tasks/AzurePowerShellV3/task.json]: M143 version `3.1.11`
M144 version `3.1.12`
Will check whats the best here.

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 [hunk 4]
```diff
@@ -643,24 +665,22 @@ function Disconnect-AzureAndClearContext {
...
+                $null = Logout-AzureRmAccount -Scope Process -ErrorAction Stop
             }
 
             if (Get-Command -Name "Clear-AzureRmContext" -ErrorAction "SilentlyContinue") {
-                Write-Host "##[command]Clear-AzureRmContext -Scope Process"
-                $null = Clear-AzureRmContext -Scope Process
-                Write-Host "##[command]Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue"
-                $null = Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
+                Write-Host "##[command]Clear-AzureRmContext -Scope Process -Scope Process -ErrorAction Stop"
+                $null = Clear-AzureRmContext -Scope Process -ErrorAction Stop
```
Comment (rajatagrawal-dev) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: Any specific reason to move from SilentlyContinue to Stop

Comment (vincentdass) [reply Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: just to catch the error message in the debug logs.

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 [hunk 5]
```diff
@@ -63,8 +63,12 @@ function Initialize-AzureSubscription {
     if ($Endpoint.Auth.Scheme -eq 'ServicePrincipal' -and !$script:azureModule -and (Get-Command -Name "Clear-AzureRmContext" -ErrorAction "SilentlyContinue")) {
         Write-Host "##[command]Clear-AzureRmContext -Scope Process"
         $null = Clear-AzureRmContext -Scope Process
-        Write-Host "##[command]Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue"
-        $null = Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
+    }
+
+    if (Get-Command -Name "Disable-AzureRmContextAutosave" -ErrorAction "SilentlyContinue") 
+    {
+        Write-Host "##[command]Disable-AzureRmContextAutosave -ErrorAction SilentlyContinue"
```
Comment (arjgupta) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: set error action to stop and put this under a try-catch

Comment (vincentdass) [reply Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: done.

**Pull request merged**